torch/export/export_schema.py

```diff
@@ -0,0 +1,326 @@
+    preserve_format = auto()
+
+
+# Copied from torch/_C/__init__.pyi.in
+@dataclass
+class Device:
+    # Defined in torch/csrc/Device.cpp
+    type: str
+    index: int
+
+
+SymInt = Union[
+    int,
+    str
+]
+
+Scalar = Union[int, float, bool]
+
+# This is a Tensor Arugment used in the args of an node
+# We intentionally don't store the tensor's storage, nor the tensor's meta data here,
+# as the same tensor argument can be used in multiple nodes, and we want to avoid storing the same data multiple times.
+# In another word, this field is an reference to the tensor, not the tensor itself.
+@dataclass
+class TensorArgument:
+    name: str   # identifier of the tensor, which must exist in graph's ivalues map
+
+
+@dataclass
+class Argument:
+
+    # Permissible types for operator's arguments
+    class ArgumentType(Enum):
+        TENSOR = auto()
+        # TENSORS = auto()    # !!! This is an important decision to decide how to handle list of tensors. More discussion to come.
+
+        SCALAR = auto()     # !!! Scalar is already an union type: Union[int, float, bool], check if serialization library can handle this
+        # SCALARS = auto()    # !!! for Scalar[], used in native_function.yaml, but not used in canonical aten ops yet... Consider if we need this
+
+        BOOL = auto()
+        BOOLS = auto()      # for bool[]
+                            # !!! There are use of bool[3] in canonical aten ops, consider if we can simplify this
+
+        INT = auto()
+        INTS = auto()       # for int[]
+
+        FLOAT = auto()
+        FLOATS = auto()     # for float[]
+
+        STRING = auto()
+        # STRINGS = auto()    # !!! There is no str[] in native_function.yaml. Consider if this is needed for expressiveness
+
+        # SYMINT = auto()    # !!! Can Symint be an arguement?, there are symint in native_function.yaml
+        # SYMINT = auto()    # !!! Can Symint[] be an arguement? there are symint[] in native_function.yaml
+
+        # GRAPH = auto()      # !!! Consider how to handle condition op, which need to pass in a graph for the branch
+        # GRAPHS = auto()      # !!! What about list of graphs? Do we need this?
+
+        # !!! Following types doesn't have a list version in native_function.yaml
+        SCALAR_TYPE = auto()
+        MEMORY_FORMAT = auto()
+        LAYOUT = auto()
+        DEVICE = auto()
+
+    # Declaration of which type the value field is storing
+    type: ArgumentType
+
+    val: Union[
+        TensorArgument,
+        # List[TensorArgument],
+        Scalar,
+        # List[Scalar],
+        bool,
+        List[bool],
+        int,
+        List[int],
+        float,
+        List[float],
+        str,
+        # List[str],
+        # SymInt,
+        # List[SymInt],
+        # Graph,
+        # List[Graph],
+        ScalarType,
+        MemoryFormat,
+        Layout,
+        Device,
+    ]
+
+@dataclass
+class KeywordArgument:
```

Comment:
seems like you are trying to schematize the function signature of a graph? Why is that needed?